tests/EntityControlTraitTest.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+namespace RonasIT\Support\Tests;
+
+use RonasIT\Support\Tests\Support\Mock\DummyConnection;
+use ReflectionMethod;
+use RonasIT\Support\Tests\Support\Mock\TestModel;
+use RonasIT\Support\Tests\Support\Mock\TestRepository;
+use RonasIT\Support\Tests\Support\Traits\MockTrait;
+use ReflectionProperty;
+
+class EntityControlTraitTest extends HelpersTestCase
+{
+    use MockTrait;
+
+    protected TestRepository $testRepositoryClass;
+    protected ReflectionProperty $onlyTrashedProperty;
+    protected ReflectionMethod $getQueryMethod;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->testRepositoryClass = new TestRepository();
+
+        $this->onlyTrashedProperty = new ReflectionProperty(TestRepository::class, 'onlyTrashed');
+        $this->onlyTrashedProperty->setAccessible('pubic');
+
+        $this->getQueryMethod = new ReflectionMethod($this->testRepositoryClass, 'getQuery');
+        $this->getQueryMethod->setAccessible('public');
+
+        DummyConnection::mock();
+    }
+
+    public function testOnlyTrashed()
+    {
+        $this->testRepositoryClass->onlyTrashed();
+
+        $onlyTrashed = $this->onlyTrashedProperty->getValue($this->testRepositoryClass);
+
+        $this->assertEquals(true, $onlyTrashed);
+    }
+
+    public function testGetQuery()
+    {
+        $sql = $this->getQueryMethod->invoke($this->testRepositoryClass)->toSql();
+
+        $this->assertEqualsFixture('get_query_sql.json', $sql);
+    }
```

Comment:
it would be great if we can test the final sql of each method instead of testing getQuery, e.g.

```php
Connection::fake();

$this->testRepository->first(['field' => 'value']);

Connection::assertQuery('select * from test_table where field = value');
```

Reply:
Please try to use https://github.com/czukowski/phpunit-mock-db it looks like what we need